sdk/core/core-amqp/src/errors.ts

```diff
@@ -551,30 +584,16 @@ export function translate(err: AmqpError | Error): MessagingError {
     return err as MessagingError;
   }
 
-  let error: MessagingError = err as MessagingError;
-
-  // OperationTimeoutError occurs when the service fails to respond within a given timeframe.
-  // Since reasons for such failures can be transient, this is treated as a retryable error.
-  if (
-    // instanceof checks on custom Errors doesn't work without manually setting the prototype within the error.
-    // Must do a name check until OperationTimeoutError is updated, and that doesn't break compatibility
-    // https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
-    (err as Error).name === "OperationTimeoutError"
-  ) {
-    error.retryable = true;
-    return error;
+  const customError = getCustomError(err);
```

Comment:
I see tests for the builtin errors like TypeError, RangeError, for amqp errors and for system errors, but none for custom errors. Please update the `errors.spect.ts` file to have a section for custom errors just like we do for the built in errors